File: docs/snippets/all/tutorials/extra_values.py
"""Log extra values with a `Points2D`."""

import dalaran as dl
import dalaran.blueprint as dlb

dl.init("dalaran_example_extra_values", spawn=True)

dl.log(
    "extra_values",
    dl.Points2D([[-1, -1], [-1, 1], [1, -1], [1, 1]]),
    dl.AnyValues(
        confidence=[0.3, 0.4, 0.5, 0.6],
    ),
)

# Set view bounds:
dl.send_blueprint(
    dlb.Spatial2DView(
        visual_bounds=dlb.VisualBounds2D(
            x_range=[-1.5, 1.5], y_range=[-1.5, 1.5]
        )
    )
)

File: docs/snippets/all/tutorials/extra_values.rs
//! Log extra values with a `Points2D`.

use std::sync::Arc;

use dalaran::external::arrow;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = dalaran::RecordingStreamBuilder::new("dalaran_example_extra_values")
        .spawn()?;

    let points = dalaran::Points2D::new([
        (-1.0, -1.0),
        (-1.0, 1.0),
        (1.0, -1.0),
        (1.0, 1.0),
    ]);
    let confidences = dalaran::AnyValues::default().with_component_from_data(
        "confidence",
        Arc::new(arrow::array::Float64Array::from(vec![0.3, 0.4, 0.5, 0.6])),
    );

    rec.log(
        "extra_values",
        &[&points as &dyn dalaran::AsComponents, &confidences],
    )?;

    Ok(())
}
